presto-spi/src/main/java/com/facebook/presto/spi/Node.java

```diff
@@ -34,4 +34,6 @@
     boolean isCoordinator();
 
     boolean isResourceManager();
+
+    String getPool();
```

Comment:
Is there a scenario where I don't want nodes to be a part of any pool. 
In other words should getPool return an Optional?

Reply:
When a node becomes the part of a cluster, it either belongs to a default pool or a specific pool if configured differently. We don't have a need for this field to be Optional.  [Here](https://github.com/prestodb/presto/pull/16670/files#diff-cd42a93bcd3f828779c6c7bc2dcf175cea53df3e136234e7877911959cbdb34dL165) is where the pool property is injected in DiscoveryNodeManager in this PR and the source is from [NodeConfig](https://github.com/prestodb/airlift/blob/master/node/src/main/java/com/facebook/airlift/node/NodeConfig.java#L70), the default value of which is "general".